python-client-migration/python_client_migration/python_client_migration/es_client_main.py

```diff
@@ -0,0 +1,63 @@
+# https://elasticsearch-py.readthedocs.io/en/v7.16.3/
+# Compatibility https://opensearch.org/docs/latest/clients/index/
+
+import os
+from random import randint
+from dotenv import load_dotenv
+from elasticsearch import Elasticsearch
+
+load_dotenv()
+
+
+def run(client, index="game-of-thrones"):
+    ## Let's start by indexing some data
+    client.index(
+        index=index,
+        body={
+            "character": "Ned Stark",
+            "quote": "Winter is coming.",
+        },
+    )
+
+    client.index(
+        index=index,
+        body={
+            "character": "Daenerys Targaryen",
+            "quote": "I am the blood of the dragon.",
+        },
+    )
+
+    client.index(
+        index=index,
+        body={
+            "character": "Tyrion Lannister",
+            "quote": "A mind needs books like a sword needs a whetstone.",
+        },
+    )
+
+    response = client.search(
+        body={
+            "query": {
+                "match": {"quote": "winter"},
+            },
+        },
+        index=index,
+    )
+
+    count = len(response["hits"]["hits"])
+    return {"success": count > 0, "count": count, "message": f"Found {count} record(s)"}
+
+
+if __name__ == "__main__":
+    # Generate random index
+    index = "game-of-thrones-" + str(randint(0, 500))
+
+    # Test ES client against ES cluster
+    ES_SERVICE_URI = os.getenv("ES_SERVICE_URI")
+    elastisearch_client = Elasticsearch(ES_SERVICE_URI, use_ssl=True)
+    print(run(elastisearch_client))
+
+    # Test OS client against ES cluster
+    OS_SERVICE_URI = os.getenv("OS_SERVICE_URI")
+    opensearch_client = Elasticsearch(OS_SERVICE_URI, use_ssl=True)
```

Comment:
Ah - this is what  I was being confused by.
We are testing and ES client (because it was created with `Elasticsearch()` against an OS backend.
So calling the variable `opensearch_client` confuses me, because clearly it is an elasticsearch client.

I *think* that instead of naming these variables as `<service>_client` it would be better to call them something like (and sorry for not having better names) `client_against_<service>` (which is like the comment) or `client_using_<service>`. 

So we might have `client_using_elasticsearch` and `client_using_opensearch`

If adopted, this change would be for all of these variables...

Reply:
I can do the proposed changes and I think they are valid. I will review the naming. However, I will do some git magic here and I will add this changes in their proper commit. Would that be okay? I need to keep the commits clean so it can be compared properly in the end. Otherwise, I think it would not work as final example.

Reply:
Yes, that sounds like an excellent idea.